packages/realm/src/OrderedCollection.ts

```diff
@@ -100,20 +95,28 @@ const PROXY_HANDLER: ProxyHandler<OrderedCollection> = {
   },
 };
 
+type OrderedCollectionInternal = binding.Results | binding.List | binding.Set;
+
 export abstract class OrderedCollection<T = unknown, EntryType extends [unknown, unknown] = [number, T]>
   extends Collection<number, T, EntryType, T, CollectionChangeCallback<T, EntryType>>
   implements Omit<ReadonlyArray<T>, "entries">
 {
   /** @internal */ protected declare realm: Realm;
-  /** @internal */ protected declare results: binding.Results;
-  /** @internal */ protected declare helpers: OrderedCollectionHelpers;
+  /** @internal */ protected declare abstract internal: OrderedCollectionInternal;
```

Comment:
Alternatively, this could be made into a template param. I tried doing that, but ran into rollup/dts issues because it requires passing a `binding` type into a template parameter but `binding` is `@internal` so rollup gets confused. And I couldn't make rollup happy with removing `@internal` on `binding` because of the way we use `declare module` in `native.d.ts`. May be worth trying again once we jettison rollup if we think the template param route is clearer.